client/ayon_core/hosts/substancepainter/api/lib.py

```diff
@@ -643,3 +640,86 @@ def _setup_prompt():
         return
 
     return project_mesh
+
+
+def convert_substance_object_to_python(subst_proj_option="default"):
+    """Function to convert substance C++ objects to python instance.
+    It is made to avoid any possible ValueError when C++ objects casting
+    as python instance.
+
+    Args:
+        subst_proj_option (str, optional): Substance project option.
+            Defaults to "default".
+
+    Raises:
+        ValueError: Raise Error when unsupported Substance
+            Project was detected
+
+    Returns:
+        python instance: converted python instance of the C++ objects.
+    """
+    if subst_proj_option == "default":
+        return substance_painter.project.ProjectWorkflow.Default
+    elif subst_proj_option == "uvTile":
+        return substance_painter.project.ProjectWorkflow.UVTile
+    elif subst_proj_option == "textureSetPerUVTile":
```

Comment:
These are the only three that resolve a lowercase first letter whereas the others match exactly with the python attribute name.

Also, I think it's not a good idea to have on function that remaps a string to attributes of different properties, e.g. `ProjectWorkflow.TextureSetPerUVTile` and `TangentSpace.PerFragment`. That's just waiting for a future update of substance painter suddenly adds `TangentSpace.Default` as well breaking the logic, but worse - calling `convert("default")` is just unclear to what of the attributes it converts.

We could also implement like this:
```python
def _convert(attr):
    value = substance_painter.project
    for key in attr.split("."):
        value = getattr(root, key, default=None)
        if value is None:
            return
    return value
```

And then make the values the string including the dots:
```python
_convert("ProjectWorkflow.TextureSetPerUVTile")
```